cli/pcluster/update/update_command.py

```diff
@@ -0,0 +1,160 @@
+        base_config = PclusterConfig(config_file=args.config_file, cluster_name=args.cluster_name)
+        try:
+            patch = ConfigPatch(base_config, target_config)
+            LOGGER.info("Found Changes:")
+            if patch.changes:
+                allowed, rows, actions_needed = patch.check()
+                print(tabulate(rows, headers="firstrow"))
+                print()
+
+                if allowed:
+                    LOGGER.info("Congratulations! The new configuration can be safely applied to your cluster.")
+                else:
+                    LOGGER.error("The new configuration cannot be safely applied to your cluster.")
+                    print("Please try the following actions:")
+                    for action in actions_needed:
+                        print("- {0}".format(action.replace("\n", "\n  ")))
+                    print("Then, retry the update.")
+                    can_proceed = False
+            else:
+                LOGGER.info("No changes found in your cluster configuration.")
+        except Exception as e:
+            LOGGER.error(e)
+            can_proceed = False
+
+    # Final consent from user
+    if can_proceed:
+        can_proceed = input("Do you want to proceed with the update? - Y/N: ").strip().lower() == "y"
+    return can_proceed
+
+
+def _autofill_cfn_params(cfn_boto3_client, args, cfn_params, stack_name, target_config):
+    cluster_section = target_config.get_section("cluster")
+
+    if cluster_section.get_param_value("scheduler") != "awsbatch":
+        # Autofill DesiredSize cfn param
+        if not args.reset_desired:
+            _restore_asg_desired_size(cfn_params, stack_name)
+    else:
+        if args.reset_desired:
+            LOGGER.info("reset_desired flag does not work with awsbatch scheduler")
+
+        # Autofill ResourcesS3Bucket cfn param
+        params = utils.get_stack(stack_name, cfn_boto3_client).get("Parameters")
+        cfn_params["ResourcesS3Bucket"] = utils.get_cfn_param(params, "ResourcesS3Bucket")
+
+
+def _restore_asg_desired_size(cfn_params, stack_name):
+    """
+    Make the ASG desired capacity to be restored to its current value after the update operation.
+
+    This is done by providing the current value to CloudFormation with the DesiredSize parameter so that the ASG will be
+    updated with that capacity instead of the minimum/initial size of the cluster.
+
+    :return:
+    """
+    asg_name = utils.get_asg_name(stack_name)
+
+    asg_settings = (
+        boto3.client("autoscaling")
+        .describe_auto_scaling_groups(AutoScalingGroupNames=[asg_name])
+        .get("AutoScalingGroups")[0]
+    )
+
+    desired_capacity = asg_settings.get("DesiredCapacity")
+
+    new_min_size = cfn_params["MinSize"]
+    new_max_size = cfn_params["MaxSize"]
+
+    if desired_capacity < new_min_size or desired_capacity > new_max_size:
+        raise Exception("The size of the Cluster cannot be ")
```

Comment:
cannot be? ;)

Reply:
Ops! :) Let's postpone this logic to a next iteration BTW